test/LevelZeroDevicePoolTest.cpp

```diff
@@ -0,0 +1,203 @@
+        EXPECT_EQ(value+sub_idx+num_accelerator_subdevice*20, m_device_pool.frequency_max(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, sub_idx, MockLevelZero::M_DOMAIN_COMPUTE));
+
+        EXPECT_EQ((uint64_t)(value+sub_idx+num_accelerator_subdevice*30), m_device_pool.active_time(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, sub_idx, MockLevelZero::M_DOMAIN_COMPUTE));
+        EXPECT_EQ((uint64_t)(value+sub_idx+num_accelerator_subdevice*40), m_device_pool.active_time_timestamp(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, sub_idx, MockLevelZero::M_DOMAIN_COMPUTE));
+
+        EXPECT_NO_THROW(m_device_pool.frequency_control(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, sub_idx, MockLevelZero::M_DOMAIN_COMPUTE, value));
+    }
+}
+
+TEST_F(LevelZeroDevicePoolTest, subdevice_conversion_error)
+{
+    const int num_accelerator = 4;
+    const int num_accelerator_subdevice = 9;
+
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR)).WillRepeatedly(Return(num_accelerator));
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP)).WillRepeatedly(Return(num_accelerator_subdevice));
+
+    LevelZeroDevicePoolImp m_device_pool(*m_levelzero);
+    GEOPM_EXPECT_THROW_MESSAGE(m_device_pool.frequency_status(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, 0, MockLevelZero::M_DOMAIN_COMPUTE), GEOPM_ERROR_INVALID, "GEOPM Requires the number of subdevices to be evenly divisible by the number of devices");
+}
+
+TEST_F(LevelZeroDevicePoolTest, domain_error)
+{
+    const int num_accelerator = 4;
+    const int num_accelerator_subdevice = 8;
+
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR)).WillRepeatedly(Return(num_accelerator));
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP)).WillRepeatedly(Return(num_accelerator_subdevice));
+
+    int dev_idx = 0;
+    int domain_count = 0; //zero to cause a throw
+    EXPECT_CALL(*m_levelzero, frequency_domain_count(dev_idx, MockLevelZero::M_DOMAIN_COMPUTE)).WillRepeatedly(Return(domain_count));
+    LevelZeroDevicePoolImp m_device_pool(*m_levelzero);
+
+    //Frequency
+    GEOPM_EXPECT_THROW_MESSAGE(m_device_pool.frequency_status(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, dev_idx, MockLevelZero::M_DOMAIN_COMPUTE), GEOPM_ERROR_INVALID, "Not supported on this hardware");
+    GEOPM_EXPECT_THROW_MESSAGE(m_device_pool.frequency_min(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, dev_idx, MockLevelZero::M_DOMAIN_COMPUTE), GEOPM_ERROR_INVALID, "Not supported on this hardware");
+    GEOPM_EXPECT_THROW_MESSAGE(m_device_pool.frequency_max(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, dev_idx, MockLevelZero::M_DOMAIN_COMPUTE), GEOPM_ERROR_INVALID, "Not supported on this hardware");
+    GEOPM_EXPECT_THROW_MESSAGE(m_device_pool.energy(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, dev_idx, MockLevelZero::M_DOMAIN_ALL), GEOPM_ERROR_INVALID, "domain "+std::to_string(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP)+" is not supported for the power domain");
+
+    //Utilization
```

Comment:
If these are not yet supported, and the tests will come in when support arrives, please open issues to that effect so we can track.  Please do that for any of these tests.  One issue encompassing all the outstanding tests is fine.

If the tests are about to come in another PR, then just ignore this request.

If there is support now, and the tests are just not written yet, that's fine too.  Just open an issue to track what's missing.

Reply:
Basic tests added.  Issues need to be filed on adding additional tests and failure cases to the LevelZero class based upon more heavily restricting the supported L0 domains for frequency, power, etc.  These checks will be largely dependent upon what is supported by upcoming firmware/hardware.